modules/image_processors/LineBoundingBoxDetector.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file   LineBoundingBoxDetector.h
+ * @brief  ライントレース用の画像処理クラス
+ * @author takuchi17 miyahara046 HaruArima08
+ */
+#include "LineBoundingBoxDetector.h"
+
+// デフォルトコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
+                                                 const cv::Scalar& _upperHSV)
+  : lowerHSV(_lowerHSV), upperHSV(_upperHSV)
+{
+  roi = cv::Rect(50, 240, 540, 240);
+  resolution = cv::Size(640, 480);
+  validateParameters();
+}
+
+// ROIを指定するオーバーロードコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& lowerHSV,
+                                                 const cv::Scalar& upperHSV, const cv::Rect& roi)
+  : lowerHSV(lowerHSV), upperHSV(upperHSV), roi(roi)
+{
+  resolution = cv::Size(640, 480);
+  validateParameters();
+}
+
+// ROIと解像度を指定するオーバーロードコンストラクタ
+LineBoundingBoxDetector::LineBoundingBoxDetector(const cv::Scalar& _lowerHSV,
+                                                 const cv::Scalar& _upperHSV, const cv::Rect& _roi,
+                                                 const cv::Size& _resolution)
+  : lowerHSV(_lowerHSV), upperHSV(_upperHSV), roi(_roi), resolution(_resolution)
+{
+  validateParameters();
+}
+
+void LineBoundingBoxDetector::validateParameters()
+{
+  // 解像度の検証：設定された解像度が定義した最小・最大値内にあるかの確認
+  if(resolution.width < ResolutionRange::MIN_WIDTH) {
+    resolution.width = ResolutionRange::MIN_WIDTH;
+  } else if(resolution.width > ResolutionRange::MAX_WIDTH) {
+    resolution.width = ResolutionRange::MAX_WIDTH;
+  }
+  if(resolution.height < ResolutionRange::MIN_HEIGHT) {
+    resolution.height = ResolutionRange::MIN_HEIGHT;
+  } else if(resolution.height > ResolutionRange::MAX_HEIGHT) {
+    resolution.height = ResolutionRange::MAX_HEIGHT;
+  }
+
+  // ROI検証：ROIの位置とサイズが解像度の枠内に収まるかの確認
+  // ROI左上座標が負の場合は0にクリップ
+  if(roi.x < 0) roi.x = 0;
+  if(roi.y < 0) roi.y = 0;
+  // ROIが解像度を超えている場合は解像度に合わせる
+  if(roi.x + roi.width > resolution.width) roi.width = resolution.width - roi.x;
+  if(roi.y + roi.height > resolution.height) roi.height = resolution.height - roi.y;
+}
+
+void LineBoundingBoxDetector::detect(const cv::Mat& frame, BoundingBoxDetectionResult& result)
+{
+  result.wasDetected = false;  // 初期状態は検出失敗
+
+  if(frame.empty()) {
+    std::cerr << "Error: Input frame is empty." << std::endl;
+    return;
+  }
+
+  // 入力画像が指定の解像度と異なる場合、リサイズ処理
+  cv::Mat frameProcessed;
```

Comment:
frame格納しとかないとまずい
```suggestion
  cv::Mat frameProcessed = frame.clone();
```
もしくは、引数にconstつけてるから下でもいい
```suggestion
  cv::Mat frameProcessed = frame;
``